pkg/controller/core.oam.dev/v1alpha2/application/apply.go

```diff
@@ -126,27 +126,39 @@ func (h *appHandler) apply(ctx context.Context, ac *v1alpha2.ApplicationConfigur
 	return nil
 }
 
-func (h *appHandler) statusAggregate(appfile *appfile.Appfile) ([]common.ApplicationComponentStatus, bool, error) {
+func (h *appHandler) statusAggregate(appFile *appfile.Appfile) ([]common.ApplicationComponentStatus, bool, error) {
 	var appStatus []common.ApplicationComponentStatus
 	var healthy = true
-	for _, wl := range appfile.Workloads {
+	for _, wl := range appFile.Workloads {
 		var status = common.ApplicationComponentStatus{
 			Name:    wl.Name,
 			Healthy: true,
 		}
-		pCtx := process.NewContext(wl.Name, appfile.Name, appfile.RevisionName)
+
+		var (
+			outputSecretName string
+			err              error
+		)
+		if wl.IsCloudResourceProducer() {
+			outputSecretName, err = appfile.GetOutputSecretNames(wl)
+			if err != nil {
+				return nil, false, errors.WithMessagef(err, "app=%s, comp=%s, setting outputSecretName error", appFile.Name, wl.Name)
+			}
+		}
+
+		pCtx := process.NewContext(h.app.Namespace, wl.Name, appFile.Name, appFile.RevisionName, outputSecretName, wl.RequiredSecrets)
```

Comment:
The code could like:

```suggestion
       pCtx := process.NewContext(h.app.Namespace, wl.Name, appFile.Name, appFile.RevisionName)
		if wl.IsCloudResourceProducer() {
			outputSecretName, err = appfile.GetOutputSecretNames(wl)
			if err != nil {
				return nil, false, errors.WithMessagef(err, "app=%s, comp=%s, setting outputSecretName error", appFile.Name, wl.Name)
			}
			pCtx.InsertSecrets(outputSecretName, wl.RequiredSecrets)
		}
```

The Code will definitely clearer after this refactor